src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -797,11 +788,9 @@ private async ValueTask<TResult> ReceiveAsyncPrivate<TResult>(Memory<byte> paylo
 
             if (NetEventSource.Log.IsEnabled()) NetEventSource.ReceiveAsyncPrivateStarted(this, payloadBuffer.Length);
 
-            CancellationTokenRegistration registration = default;
+            CancellationTokenRegistration registration = cancellationToken.Register(static s => ((ManagedWebSocket)s!).Abort(), this);
```

Comment:
If the token already has cancellation requested when Register is called, it'll invoke the callback synchronously. If that callback throws, the exception will propagate out. Is there any situations where Abort will throw an exception and we want the catch/finally blocks from this try to deal with it? If yes, this is changing behavior.

Reply:
> Is there any situations where Abort will throw an exception

Abort only sets the state and calls Dispose... I'd assume if WebSocket Dispose is throwing, there's something really really wrong. That being said, a candidate for a potential dispose exception would be a user-provided stream. So I agree this will be technically changing.

Even though I don't really see anyone depending on an exeption-on-dispose stream, I guess we'd better play safe and leave Register within the try-catch